nnoir-onnx/nnoir_onnx/operators/mat_mul.py

```diff
@@ -9,7 +18,7 @@ def __init__(self, node, *args):
 
     def to_function(self, env, constants):
         [x, W] = self.node.input
-        if W in constants:
+        if W in constants and len(constants[W].shape) == 2 and np.prod(env[x].shape[1:]) == constants[W].shape[1]:
```

Comment:
MatMul should not be "chainer like" Linear.

Reply:
What do you mean?

Reply:
```
>>> x.shape
(2, 3, 4, 5)
>>> W.shape
(5, 6)
>>> np.matmul(x,W).shape
(2, 3, 4, 6)
```

Reply:
I cannot catch what you want to say.
Your example is not sufficient the condition check pointed by you.

Can you clarify what you want to say in words.

Reply:
Sorry, my example is confusing.

At first, you should add and try this test case.
```python
def test_matmul_06():
    '''
    opset version >= 9
    '''

    shape = (3, 1, 1)
    w_shape = (1, 1)
    out_shape = (3, 1, 1)

    class MatMulTester(Base):
        def create_onnx(self) -> onnx.ModelProto:
            node = make_node("MatMul", inputs=["x", "W"], outputs=["y"])
            inputs = [info("x", TensorProto.FLOAT, shape)]
            outputs = [info("y", TensorProto.FLOAT, out_shape)]

            init_W = from_array(np.random.rand(*w_shape).astype(np.float32), "W")

            graph = make_graph([node], "add_graph", inputs, outputs, initializer=[init_W])
            model = make_model(graph)
            return model

    x = np.random.rand(*shape).astype(np.float32)

    outputs = ["y"]
    MatMulTester({"x": x}, outputs).run()
```

`W in constants and len(constants[W].shape) == 2 and np.prod(env[x].shape[1:]) == constants[W].shape[1]` is wrong condition. We can convert MatMul(x, W) to single Linear(x, W.T) when x.ndim == W.ndim == 2.

Reply:
Thx. I will fix my code.